battle-engine.js

```diff
@@ -197,53 +220,22 @@ BattlePokemon = (() => {
 		this.clearVolatile(true);
 	}
 
-	BattlePokemon.prototype.trapped = false;
-	BattlePokemon.prototype.maybeTrapped = false;
-	BattlePokemon.prototype.maybeDisabled = false;
-	BattlePokemon.prototype.hp = 0;
-	BattlePokemon.prototype.maxhp = 100;
-	BattlePokemon.prototype.illusion = null;
-	BattlePokemon.prototype.fainted = false;
-	BattlePokemon.prototype.faintQueued = false;
-	BattlePokemon.prototype.lastItem = '';
-	BattlePokemon.prototype.ateBerry = false;
-	BattlePokemon.prototype.status = '';
-	BattlePokemon.prototype.position = 0;
-
-	BattlePokemon.prototype.lastMove = '';
-	BattlePokemon.prototype.moveThisTurn = '';
-
-	BattlePokemon.prototype.lastDamage = 0;
-	BattlePokemon.prototype.lastAttackedBy = null;
-	BattlePokemon.prototype.usedItemThisTurn = false;
-	BattlePokemon.prototype.newlySwitched = false;
-	BattlePokemon.prototype.beingCalledBack = false;
-	BattlePokemon.prototype.isActive = false;
-	BattlePokemon.prototype.isStarted = false; // has this pokemon's Start events run yet?
-	BattlePokemon.prototype.transformed = false;
-	BattlePokemon.prototype.duringMove = false;
-	BattlePokemon.prototype.hpType = 'Dark';
-	BattlePokemon.prototype.hpPower = 60;
-	BattlePokemon.prototype.speed = 0;
-	BattlePokemon.prototype.abilityOrder = 0;
-
-	BattlePokemon.prototype.toString = function () {
+	toString() {
 		let fullname = this.fullname;
 		if (this.illusion) fullname = this.illusion.fullname;
 
 		let positionList = 'abcdef';
 		if (this.isActive) return fullname.substr(0, 2) + positionList[this.position] + fullname.substr(2);
 		return fullname;
-	};
-	// "static" function
-	BattlePokemon.getDetails = function (side) {
+	}
+	static getDetails(side) {
```

Comment:
I think it's because it can be overwritten by something, and it falls back on the static method with this bound.